### ISSUE TITLE
.CSS / .SCSS extension conflict when from the link tag helper when EnableTagHelperBundling = false and using AddScssBundle()

### ISSUE THREAD

**Issue by tommytayman** [ligershark/WebOptimizer.Sass#70]
### Background

We're in the process of replacing the Bundle & Minify plug-in with this library since it gives us better bundle flexibility in addition to compiling .sass files.  

### Setup
- Visual Studio 2022 - 17.8.3
- .NET - 8.0 
- LigerShark.WebOptimizer.Core - 3.0.396
- LigerShark.WebOptimizer.Sass - 3.0.102

### Test Setup

1.  Open the sample project in this repository and change the Startup.cs from this:

```
 services.AddWebOptimizer(pipeline =>
{
    // Creates a Sass bundle. Globbing patterns supported.
    pipeline.AddScssBundle("/css/bundle.css", "scss/*.scss");

    // This will compile any Sass file that isn't part of any bundle
    pipeline.CompileScssFiles();
});
```

to this

```
services.AddWebOptimizer(pipeline =>
{
    // Creates a Sass bundle. Globbing patterns supported.
    pipeline.AddScssBundle("/css/bundle.css", "scss/*.scss");

    // This will compile any Sass file that isn't part of any bundle
    pipeline.CompileScssFiles();
}, option =>
{
    option.EnableTagHelperBundling = false;
});
```

###Expected Results

Since we want each file to render instead of the single bundle, we would expected

```
<link href="scss/a.scss?v=Hash" rel="stylesheet" />
<link href="scss/b.scss?v=Hash" rel="stylesheet" />
```
This is expected and works due to the request to render simply the individual files that make up this bundle and because this library will render a SCSS file as a minified CSS file upon request.

### Actual Results

<link href="scss/a.css" rel="stylesheet" />
<link href="scss/b.css" rel="stylesheet" />

We can see that the extension has been changed from the link tag helper from `.scss` to `.css` resulting in a 404 error on the server and no styles being rendered.  Also note the lack of the ?v=Hash on these elements.  The same issue affects the index page which directly references the `/scss/a.scss` file [here](https://github.com/ligershark/WebOptimizer.Sass/blob/c659ac82182457c5d6b8736043a9366742334e70/sample/WebOptimizer.Sass.Sample/Pages/Index.cshtml#L7) when the `EnableTagHelperBundling=false` is set.

### Potential Solution

I cloned this repository as well as the `WebOptimizer.Core` repository.  Next, I removed the NuGet reference from this project and referenced the local copy of the Core project directly.  When I comment out this [line of code](https://github.com/ligershark/WebOptimizer/blob/e3673d083f165cb0d90e9624b8279a7239610217/src/WebOptimizer.Core/Taghelpers/LinkTagHelper.cs#L103) the project returns the expected values in both the "un-budling" (render each file individually from a bundle reference) scenario on the Bundle.cshtml page and the direct reference on the Index.cshtml page.  

I am not familiar enough with this project to know what potential impacts removing that line of code would have, but it does seem to conflict with this library and un-bundling .scss files

**Comment by eademi** [comment]
We have the same exact issue! Were you able to find a more permanent solution?

**Comment by tommytayman** [comment]
@eademi - Sorry, no.  We're not using `.scss` quite yes but have plans to in the near future.  Knowing that we would be doing some `.scss` work is what initially prompted me to update our bundling solution.  The only way that I have found is just to leave `EnableTagHelperBundling` to true in local development which makes tracking down JS errors a bit more problematic.  

I'm hoping the library authors will be able to patch this up prior to us getting to `.scss` processing.  I would submit a PR with my proposed fix, but not sure if there are other implications of removing that line.

**Comment by eademi** [comment]
@tommytayman We ended up using another package (AspNetCore.SassCompiler) for compiling .scss into .css and we only use WebOptimizer.Core for bundling and minifying.  Thanks again!

**Comment by tommytayman** [comment]
@eademi - Nice!  May look into this as well

**Comment by AJKKarthik024** [comment]
yes same issue,  3.0.99 version is fine, so something changed in last version

**Comment by AJKKarthik024** [comment]
@eademi @tommytayman what is the issue EnableTagHelperBundling =true, i don't see any issue with it, what problems we have in JS debugging, I am intending to use this extension, are there any problems to use in prod,and what advanytages does AspNetCore.SassCompiler have over weboptimizer.saas, i find weboptimizer.saas more easy

**Comment by tommytayman** [comment]
@AJKKarthik024 

> what is the issue with EnableTagHelperBundling =true, what problems we have in JS debugging

this does not impact anything JS related nor `EnableTagHelperBundling=true`.  The issue is around .scss files and when 'EnableTagHelperBundling=false'.  When you have this set to false, each styling file is listed individually but `WebOptimizer` is changing the `.scss` extension to `.css` which causes a 404 error.  If the extension was not changed, then `.scss` files would render correctly both when bundling and when not bundling.

>are there any problems to use in prod,

None that I've seen.  We've been using this for a month or so now with no issues.

> and what advantages does AspNetCore.SassCompiler have over weboptimizer.saas,

I can't speak to this as I have no experience with the other compiler.

**Comment by AJKKarthik024** [comment]
@tommytayman ,i have one question, is there any way that when i place multiple scss files in layout
like below
<link rel="stylesheet" href="~/templates/assets-temp1/css/styles/header/main-header-option1.scss"  defer />
<link rel="stylesheet" href="~/templates/assets-temp1/css/styles/header/sub-header-option1.scss"  defer />

in multiple partial views those gets bundled dynamically based on all the sass files into single css and return, instead of multiple network calls

**Comment by pauldbentley** [comment]
I am facing this issue today as well and noticed the line of code you highlighted.  My use-case is exactly the same as yours.  In development I want each scss file listed out from the taghelper but the file extension gets changed.  I have tried playing with a custom IFileProvider but can't seem to get anything to work.

**Comment by MWin10** [comment]
I ended up in implementing an extension method that registers the missing routes by taking each source file and replace the extension `.scss` with `.css`
```C#
    internal static class WebOptimizerExtender
    {
        internal static void AddCustomScssBundle(this WebOptimizer.IAssetPipeline pipeline, string route, params string[] sourceFiles)
        {            
            foreach (string sourceFile in sourceFiles)
            {
               // add single source file as "bundle"
                string cssRoute = sourceFile.Replace(".scss", ".css", StringComparison.OrdinalIgnoreCase);
                pipeline.AddScssBundle(cssRoute, sourceFile);
            }

            // add the original (complete) bundle 
            pipeline.AddScssBundle(route, sourceFiles);
        }
    }
```
 
This could combined with passing `IWebHostEnvironment` to add the "single bundes" only in `Development` environment for example
